src/components/victory-gauge.js

```diff
@@ -302,75 +321,176 @@ export default class VictoryGauge extends React.Component {
       .sort(null)
       .startAngle(degreesToRadians(props.startAngle))
       .endAngle(degreesToRadians(props.endAngle))
-      .padAngle(degreesToRadians(props.padAngle))
-      .value((datum) => { return datum.y; });
+      .padAngle(degreesToRadians(props.padAngle));
+  }
+  getSegments(props) {
+    const {segments} = props;
+    if (segments && segments.length) {
+      return segments;
+    }
+    // if there are no segments provide set of data as a
+    // segment that spans entire arc. array of 1?
+    return [1];
+  }
+  getGaugeRange(props, segmentLocations) {
+    const radiansToDegrees = (r) => r * (180 / Math.PI);
+    const {domain} = props;
+    return {
+      minimum: {
+        value: domain && domain[0] || segmentLocations[0].data,
+        degrees: radiansToDegrees(segmentLocations[0].startAngle)
+      },
+      maximum: {
+        value: domain && domain[1] || segmentLocations[1].data,
+        degrees: radiansToDegrees(segmentLocations.reverse()[0].endAngle)
```

Comment:
Yes, `segments` should represent the endpoints. So you can take the difference of adjacent pairs and use those values in your existing pie layout function. If a domain is given and its extent is greater than the extent of the `segments` array, you should push the domain values onto that array and sort it before taking the differences between pairs. If `domain` is given it should always "win" in terms of what the domain of the resulting gauge actually is. Otherwise you should try to reconcile `tickValues` and `segments` and form a domain based on the maximum extent of all of both. `tickValues` are placed independently of segments, so the `tickValues` don't need to line up with segment divisions (unless of course the values are the same).